conda_build/cli/logging.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (C) 2014 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+from __future__ import annotations
+
+import logging
+import logging.config
+import os
+import os.path
+import sys
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from conda.base.context import context
+from yaml import safe_load
+
+if TYPE_CHECKING:
+    from logging import LogRecord
+
+
+# https://stackoverflow.com/a/31459386/1170370
+class LessThanFilter(logging.Filter):
+    def __init__(self, exclusive_maximum: int, name: str = "") -> None:
+        super().__init__(name)
+        self.max_level = exclusive_maximum
+
+    def filter(self, record: LogRecord) -> bool:
+        return record.levelno < self.max_level
+
+
+class GreaterThanFilter(logging.Filter):
+    def __init__(self, exclusive_minimum: int, name: str = "") -> None:
+        super().__init__(name)
+        self.min_level = exclusive_minimum
+
+    def filter(self, record: LogRecord) -> bool:
+        return record.levelno > self.min_level
+
+
+class DuplicateFilter(logging.Filter):
+    msgs: set[str] = set()
+
+    def filter(self, record: LogRecord) -> bool:
+        try:
+            return record.msg not in self.msgs
+        finally:
+            self.msgs.add(record.msg)
+
+
+def init_logging() -> None:
+    """
+    Default initialization of logging for conda-build CLI.
+
+    When using conda-build as a CLI tool (not as a library) we wish to limit logging to
+    avoid duplication and to otherwise offer some default behavior.
+    """
+    # undo conda messing with the root logger
+    logging.getLogger(None).setLevel(logging.WARNING)
```

Comment:
To remove this we will first need to remove `initialize_root_logger` from conda's log initialization (and then wait some number of conda releases so we can bump up conda-build's minimum conda dependency): https://github.com/conda/conda/blob/eb459543b0c280011216f2f22eedc1c8dcbd4d9f/conda/gateways/logging.py#L142-L147